### PR TITLE
PolyhedralGeometry: Cones of fan returns incidence matrix

### PR DESCRIPTION
**Pull request by HereAround**
cc @lkastner

### CODE REVIEW

File: src/PolyhedralGeometry/PolyhedralFan/properties.jl [hunk 1]
```diff
@@ -166,19 +166,50 @@ _ray_indices(::Val{_cone_of_dim}, PF::Polymake.BigObject; c_dim::Int = 0) = Poly
...
+[5]
+[3]
+[7]
+[2]
+[6]
+[4]
+[8]
 ```
 """
 function cones(PF::_FanLikeType{T}) where T<:scalar_types
```
Comment (lkastner) [inline src/PolyhedralGeometry/PolyhedralFan/properties.jl]:
```suggestion
function cones(PF::_FanLikeType)
```
`T` should be irrelevant here.

File: src/PolyhedralGeometry/PolyhedralFan/properties.jl [hunk 2]
```diff
@@ -226,6 +257,23 @@ julia> n_maximal_cones(PF)
...
+The cones given in this construction are non-redundant. There are six
+cones in this fan.
+```jldoctest
+julia> PF = PolyhedralFan([1 0; 0 1; -1 -1], IncidenceMatrix([[1, 2], [3]]));
+
+julia> n_cones(PF)
+6
+```
+"""
+n_cones(PF::_FanLikeType) where T<:scalar_types = nrows(cones(PF))
```
Comment (lkastner) [inline src/PolyhedralGeometry/PolyhedralFan/properties.jl]:
```suggestion
n_cones(PF::_FanLikeType) = nrows(cones(PF))
```
Same here.

**Review by lkastner — APPROVED** [review]

File: src/PolyhedralGeometry/PolyhedralFan/properties.jl (hunk 1 above)
Comment (HereAround) [reply src/PolyhedralGeometry/PolyhedralFan/properties.jl]: Thx. I have rebased (and squashed) our changes and integrated this.

File: src/PolyhedralGeometry/PolyhedralFan/properties.jl (hunk 2 above)
Comment (HereAround) [reply src/PolyhedralGeometry/PolyhedralFan/properties.jl]: I have taken care of this - see previous suggestion.

File: src/AlgebraicGeometry/ToricVarieties/AlgebraicCycles/special_attributes.jl [hunk 3]
```diff
@@ -78,8 +78,8 @@ julia> length(gens_of_rational_equivalence_classes(p2))
 """
 @attr Vector{MPolyQuoRingElem{QQMPolyRingElem}} function gens_of_rational_equivalence_classes(v::AbstractNormalToricVariety)
     g = gens(chow_ring(v))
-    r_list = [rays(c) for c in cones(v)]
-    return [prod([g[findfirst(x->x==r, rays(v))] for r in rs]) for rs in r_list]
+    cs = cones(v)
+    return [prod([g[l]^(cs[k,l]) for l in 1:nrays(v)]) for k in 1:n_cones(fan(v))]
 end
```
Comment (lkastner) [inline src/AlgebraicGeometry/ToricVarieties/AlgebraicCycles/special_attributes.jl]: I suggest something like
```
cr = chow_ring(v)
R = base_ring(cr)
cs = cones(v)
return [cr(R([1], cs[k,:])) for k in 1:n_cones(v)]
```

File: src/AlgebraicGeometry/ToricVarieties/AlgebraicCycles/special_attributes.jl [hunk 4]
```diff
@@ -101,11 +101,11 @@ julia> length(map_gens_of_chow_ring_to_cox_ring(p2))
...
     g2 = gens(cox_ring(v))
-    r_list = [rays(c) for c in cones(v)]
+    cs = cones(v)
     mapping = Dict{QQMPolyRingElem, MPolyDecRingElem{QQFieldElem, QQMPolyRingElem}}()
-    for rs in r_list
-      p1 = prod([g[findfirst(x->x==r, rays(v))] for r in rs]).f
-      p2 = prod([g2[findfirst(x->x==r, rays(v))] for r in rs])
+    for k in 1:nrows(cs)
+      p1 = prod([g[l]^(cs[k,l]) for l in 1:nrays(v)]).f
+      p2 = prod([g2[l]^(cs[k,l]) for l in 1:nrays(v)])
```
Comment (lkastner) [inline src/AlgebraicGeometry/ToricVarieties/AlgebraicCycles/special_attributes.jl]: As above, please avoid using `prod` and construct the monomials directly. Then there will be no multiplication which for this step should be much faster. Probably not so relevant for the overall performance.

**Review by lkastner — APPROVED** [review]

File: src/AlgebraicGeometry/ToricVarieties/AlgebraicCycles/special_attributes.jl (hunk 3 above)
Comment (HereAround) [reply src/AlgebraicGeometry/ToricVarieties/AlgebraicCycles/special_attributes.jl]: Thank you for the pointer and the discussion on slack @lkastner . I have just pushed the changes that we discussed. Cf. https://github.com/oscar-system/Oscar.jl/pull/2244/commits/03eb3727e6718b31d4f8b688539f17d418cc1fda.

File: src/AlgebraicGeometry/ToricVarieties/AlgebraicCycles/special_attributes.jl (hunk 4 above)
Comment (HereAround) [reply src/AlgebraicGeometry/ToricVarieties/AlgebraicCycles/special_attributes.jl]: Same as above, contained in https://github.com/oscar-system/Oscar.jl/pull/2244/commits/03eb3727e6718b31d4f8b688539f17d418cc1fda.

**Pull request merged**